### PR TITLE
Circleci 2 0

### PR DESCRIPTION
**Pull request by jgagnaire**

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -7,18 +7,16 @@
 
 [tox]
 envlist = py27-swift{1.13.1,2.1.0,2.2.0,2.3.0,2.5.0,2.7.1,2.10.1,2.13.1,2.15.1}
-          py27-swifthead
           pep8
 minversion = 1.8.1
 
 [testenv]
-deps =
+deps = -r{toxinidir}/test-requirements.txt
```
Comment (NicolasT) [inline tox.ini]: Why is this required?

File: tox.ini [hunk 2]
```diff
@@ -7,18 +7,16 @@
 
 [tox]
 envlist = py27-swift{1.13.1,2.1.0,2.2.0,2.3.0,2.5.0,2.7.1,2.10.1,2.13.1,2.15.1}
-          py27-swifthead
```
Comment (NicolasT) [inline tox.ini]: Any way to keep `head` working? Maybe set it as 'may fail'?

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,70 @@
+version: 2
+
+jobs:
+  build:
+    docker:
+      - image: jgagnaire/scalitysproxydswift
```
Comment (NicolasT) [inline .circleci/config.yml]: What's in this image?

Comment (jgagnaire) [reply .circleci/config.yml]: I've put all the dependencies needed in the image, based on xenial. I've added the Dockerfile content in the description of
https://hub.docker.com/r/jgagnaire/scalitysproxydswift/

Comment (NicolasT) [reply .circleci/config.yml]: Then I'd prefer to have the `Dockerfile` in this repository (not in the root, of course, but some suitable dir), and set up a DockerHub 'autobuild' image using it.

File: tox.ini (hunk 1 above)
Comment (jgagnaire) [reply tox.ini]: No real modification here, it was included line 37 below "swifthead:". I just moved it there for more readability after removing the "swifthead:" line below

File: .circleci/config.yml (hunk 3 above)
Comment (jgagnaire) [reply .circleci/config.yml]: Done, I have added the Dockerfile to the .circleci/ directory, and configured a public automated build on Docker hub

File: tox.ini (hunk 2 above)
Comment (jgagnaire) [reply tox.ini]: Hm as far as I can see in the tox documentation, it seems that a command can be set as 'may fail' by appending a dash (-) before it, but I haven't found information related to envlist elements that can be set the same way.

As it will be restored in the PR merging the queens_compat branch, maybe I can just comment the swifthead lines instead of removing them I you wish?

File: .circleci/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    libtool \
+    zlib1g-dev \
+    python2.7 \
+    python-dev \
+    python-pip \
+    python-tox
+
+RUN pip install coverage
+
+RUN git clone https://github.com/openstack/liberasurecode.git && \
```
Comment (xroche-scality) [inline .circleci/Dockerfile]: Comment why lib sucks on Ubuntu

Comment (jgagnaire) [reply .circleci/Dockerfile]: The version installed on ubuntu by "apt-get install liberasurecode-dev" is 1.1.0-3 on xenial, which contains a bug related to dependencies on internal openstack libraries libisal and libshss.

This bug is corrected in versions >= 1.2.1, but apt-get is not aware of them:
"E: Version '1.3.0' for 'liberasurecode-dev' was not found"

Building and installing liberasurecode-dev from source solves the problem.


More information:

https://stackoverflow.com/questions/40844556/compile-liberasurecode-missing-libshss-so

https://ask.openstack.org/en/question/93267/unable-to-start-swift-proxy-liberasurecode-missing-libshssso/
